lib/ansible/modules/network/cnos/cnos_vrf.py

```diff
@@ -0,0 +1,333 @@
+  type: list
+  sample:
+    - vrf context test
+    - rd 1:100
+    - interface Ethernet1
+    - vrf member test
+"""
+import re
+import time
+
+from copy import deepcopy
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.common.utils import remove_default_spec
+from ansible.module_utils.network.cnos.cnos import load_config, run_commands
+from ansible.module_utils.network.cnos.cnos import cnos_argument_spec, check_args
+
+
+def search_obj_in_list(name, lst):
+    for o in lst:
+        if o['name'] == name:
+            return o
+
+
+def map_obj_to_commands(updates, module):
+    commands = list()
+    want, have = updates
+    state = module.params['state']
+    purge = module.params['purge']
+
+    for w in want:
+        name = w['name']
+        rd = w['rd']
+        interfaces = w['interfaces']
+
+        obj_in_have = search_obj_in_list(name, have)
+
+        if name == 'default':
+            module.fail_json(msg='VRF context default is reserved')
+        elif len(name) > 63:
+            module.fail_json(msg='VRF name is too long')
+        if state == 'absent':
+            if name == 'management':
+                module.fail_json(msg='Management VRF context cannot be deleted')
+            if obj_in_have:
+                commands.append('no vrf context %s' % name)
+        elif state == 'present':
+            if not obj_in_have:
+                commands.append('vrf context %s' % name)
+
+                if rd is not None:
+                    commands.append('rd %s' % rd)
+
+                if w['interfaces']:
+                    for i in w['interfaces']:
+                        commands.append('interface %s' % i)
+                        commands.append('vrf member %s' % w['name'])
+            else:
+                if w['rd'] is not None and w['rd'] != obj_in_have['rd']:
+                    commands.append('vrf context %s' % w['name'])
+                    commands.append('rd %s' % w['rd'])
+
+                if w['interfaces']:
+                    if not obj_in_have['interfaces']:
+                        for i in w['interfaces']:
+                            commands.append('interface %s' % i)
+                            commands.append('vrf member %s' % w['name'])
+                    elif set(w['interfaces']) != obj_in_have['interfaces']:
+                        missing_interfaces = list(set(w['interfaces']) - set(obj_in_have['interfaces']))
+
+                        for i in missing_interfaces:
+                            commands.append('interface %s' % i)
+                            commands.append('vrf member %s' % w['name'])
+
+    if purge:
+        for h in have:
+            obj_in_want = search_obj_in_list(h['name'], want)
+            if not obj_in_want:
+                commands.append('no vrf context %s' % h['name'])
+
+    return commands
+
+
+def map_config_to_obj(module):
+    objs = []
+    output = run_commands(module, {'command': 'show vrf'})
```

Comment:
please check if output is not `None` before operating on the same.

Reply:
Done